### ISSUE TITLE
Logstash Plugin Generator Allows creationg of plugin without setting --type

### ISSUE THREAD

**Issue by talevy** [elastic/logstash#5412]
new logstash plugin generator errors out when creating new plugin

```
~/Code/es/talevy-logstash (master)$ bin/logstash-plugin generate --name codec-confluent-avro
 Creating /Users/tal/Code/es/talevy-logstash/logstash--codec-confluent-avro
ERROR: Plugin creation Aborted, message: No such file or directory - No such directory: /Users/tal/Code/es/talevy-logstash/lib/pluginmanager/templates/-plugin
```

Expected:

- no error
- and creating `logstash-codec-confluent-avro` instead of `logstash--codec-confluent-avro`

**Comment by talevy** [comment]
btw, this is being run from `master` after a `rake bootstrap`

**Comment by talevy** [comment]
looks like it runs even though `--type` is not set

**Comment by talevy** [comment]
oh. and it doesn't support codecs

**Comment by purbon** [comment]
thanks for testing this features @talevy, I'm sorry for your pain. Yes as you described codecs are not supported still, but are on it's way very soon.

I see a couple of errors to fix here.

* Required parameters are not properly validates.
* Add codecs to the set of plugins we can create.

- purbon

**Comment by purbon** [comment]
This two issues should already be fixed in #5419 and #5421, sorry for your issues and thanks a lot for checking it.

**Comment by purbon** [comment]
Fixed in #5419 and #5421

**Comment by talevy** [comment]
awesome! thanks @purbon. I will test it today
